lib/ui/src/components/sidebar/Sidebar.tsx

```diff
@@ -100,28 +100,24 @@ export const Sidebar: FunctionComponent<SidebarProps> = React.memo(
     enableShortcuts = true,
     refs = {},
   }) => {
+    const collapseFn = DOCS_MODE ? collapseAllStories : collapseDocsOnlyStories;
     const selected: Selection = useMemo(() => storyId && { storyId, refId }, [storyId, refId]);
-    const stories = useMemo(
-      () => (DOCS_MODE ? collapseAllStories : collapseDocsOnlyStories)(storiesHash),
```

Comment:
The collapse functions actually remove keys from `storiesHash`. Due to the way the `adaptedRefs` was written, those keys never got removed, causing the nesting issue.